mypy/checker.py

```diff
@@ -4378,6 +4380,61 @@ def infer_issubclass_maps(self, node: CallExpr,
         yes_map, no_map = map(convert_to_typetype, (yes_map, no_map))
         return yes_map, no_map
 
+    def check_long_tuple_assignment(self,
+                               subtype: ProperType,
+                               supertype: ProperType,
+                               context: Context,
+                               msg: str = message_registry.INCOMPATIBLE_TYPES,
+                               subtype_label: Optional[str] = None,
+                               supertype_label: Optional[str] = None,
+                               code: Optional[ErrorCode] = None) -> bool:
+        """Check if assignment error using long tuple initializers."""
+
+        def format_error_msg(msg: str, lhs_types: List[Type], rhs_types: List[Type]) -> str:
```

Comment:
Style nit: We generally avoid nested functions (unless they are very short). It would be more consistent if this function was a method (or maybe a module-level function, but this seems so tightly coupled to the surrounding method that it may not be a good idea).

Reply:
The current implementation uses two nested functions: `format_long_tuple_type`, `format_error_msg`, should I turn all two of them into class methods(maybe a more clear name for latter one)?

Reply:
Our style is to generally use instance methods even if `self` is not used unless the method is being called from outside the class as `ClsName.method(...)`.